lib/mix/lib/mix/tasks/profile.cprof.ex

```diff
@@ -0,0 +1,333 @@
+    process_config opts
+
+    # Start app after rewriting System.argv,
+    # but before requiring and evaling
+    unless "--no-start" in args do
+      Mix.Task.run "app.start", args
+    end
+    process_load opts
+
+    _ = if file do
+      if File.regular?(file) do
+        profile_code(File.read!(file), opts)
+      else
+        Mix.raise "No such file: #{file}"
+      end
+    end
+
+    :ok
+  end
+
+  defp process_config(opts) do
+    Enum.each opts, fn
+      {:config, value} ->
+        Mix.Task.run "loadconfig", [value]
+      _ ->
+        :ok
+    end
+  end
+
+  defp process_load(opts) do
+    require_runner =
+      if opts[:parallel] do
+        &Kernel.ParallelRequire.files/1
+      else
+        fn(files) -> Enum.each(files, &Code.require_file/1) end
+      end
+
+    Enum.each opts, fn
+      {:require, value} ->
+        case filter_patterns(value) do
+          [] ->
+            Mix.raise "No files matched pattern #{inspect value} given to --require"
+          filtered ->
+            require_runner.(filtered)
+        end
+      {:eval, value} ->
+        profile_code(value, opts)
+      _ ->
+        :ok
+    end
+  end
+
+  defp filter_patterns(pattern) do
+    Path.wildcard(pattern)
+    |> Enum.uniq
+    |> Enum.filter(&File.regular?/1)
+  end
+
+  # Profiling functions
+
+  defp profile_code(code_string, opts) do
+    content = 
+      quote do
+        unquote(__MODULE__).profile(fn ->
+          unquote(Code.string_to_quoted!(code_string))
+        end, unquote(opts))
+      end
+    # Use compile_quoted since it leaves less noise than eval_quoted
+    Code.compile_quoted(content)
+  end
+
+  @doc false
+  def profile(fun, opts) do
+    fun
+    |> profile_and_analyse(opts)
+    |> print_output
+
+    :cprof.stop()
+  end
+
+  defp profile_and_analyse(fun, opts) do
+    if Keyword.get(opts, :warmup, true) do
+      IO.puts "Warmup..."
+      fun.()
+    end
+
+    num_matched_functions = case Keyword.get(opts, :func_spec) do
+      nil -> :cprof.start()
+      func_spec ->
+        case parse_func_spec(func_spec) do
+          :on_load -> :cprof.start({:on_load})
+          {module} -> :cprof.start(module)
+          {module, function} -> :cprof.start(module, function)
+          {module, function, arity} -> :cprof.start(module, function, arity)
+        end
+      end
+
+    apply(fun, [])
+
+    :cprof.pause() 
+
+    limit  = Keyword.get(opts, :limit) 
+    module = Keyword.get(opts, :module)
+
+    analysis_result = case {limit, module} do
+      {nil, nil} -> :cprof.analyse()
+      {limit, nil} -> :cprof.analyse(limit)
+      {limit, module} -> 
+        module_atom = string_to_existing_module_atom(module)
+        case limit do
+          nil -> :cprof.analyse(module_atom)
+          _   -> :cprof.analyse(module_atom, limit)
+        end
+    end
+
+    {num_matched_functions, analysis_result}
+  end
+
+  defp string_to_existing_module_atom(":'Elixir'." <> module), do: String.to_existing_atom(module)
```

Comment:
I don't think this clause will ever be invoked, you can likely remove it. I also don't think you should use String.to_existing_atom here or in the clauses below as it will crash if the module was not loaded (because it never appeared in the profiler, for example).